internal/alb/ls/listener.go

```diff
@@ -250,3 +264,76 @@ func (controller *defaultController) buildDefaultActions(ctx context.Context, op
 	}
 	return buildActions(ctx, authCfg, options.IngressAnnos, backend, options.TGGroup)
 }
+
+// inferCertARNs retrieves a set of certificates from ACM that matches the ingress' hosts list
+func inferCertARNs(acmsvc aws.ACMAPI, ingress *extensions.Ingress, logger *log.Logger) ([]string, error) {
+	var certArns []string
+	var seen = map[string]bool{}
+	var ingressHosts = uniqueHosts(ingress)
+
+	certs, err := acmsvc.ListCertificates([]string{acm.CertificateStatusIssued})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range certs {
+		for _, h := range ingressHosts {
+			if domainMatchesHost(aws.StringValue(c.DomainName), h) {
+				logger.Infof("Domain name '%s', matches TLS host '%v', adding to Listener", aws.StringValue(c.DomainName), h)
+				if !seen[aws.StringValue(c.CertificateArn)] {
+					certArns = append(certArns, aws.StringValue(c.CertificateArn))
+					seen[aws.StringValue(c.CertificateArn)] = true
+				}
+			} else {
+				logger.Debugf("Ignoring domain name '%s', doesn't match '%s'", aws.StringValue(c.DomainName), h)
+			}
+		}
+	}
+
+	return certArns, nil
+}
+
+func domainMatchesHost(domainName string, tlsHost string) bool {
+	if strings.HasPrefix(domainName, "*.") {
+		ds := strings.Split(domainName, ".")
+		hs := strings.Split(tlsHost, ".")
+
+		if len(ds) != len(hs) {
+			return false
+		}
+
+		for i, dp := range ds {
```

Comment:
i think it for-loop can be simplied to  ```return reflect.DeepEqual(ds[1:],hs[1:])```

Reply:
That would cause it to erroneously return true for the following trivial case:
```
ds := []string{"google", "com"}
hs := []string{"facebook", "com"}
```

Reply:
i don't think so, since you already have two precondition before:

1. strings.HasPrefix(domainName, "*."), ensures ds begins with "*",
1. if len(ds) != len(hs), ensures hs is at least 1. (so hs[1:] is safe).

use reflect.DeepEqual(ds[1:], hs[1:]) after the two condition seems easier to read 🤣

Reply:
oh whoops, I missed the pre-condition. Good catch.

Reply:
Oh I see, `strings.Split("*.", ".")` actually evaluates to `["*", ""]` which is size 2, so if `ds` and `hs` are equal then they are both at least size 2 and `ds[1:]` and `hs[1:]` will be safe operations.